test/python/basicaer/test_qasm_simulator.py

```diff
@@ -40,6 +50,31 @@ def setUp(self):
         transpiled_circuit.name = 'test'
         transpiled_circuit = transpile(transpiled_circuit, backend=self.backend)
         self.qobj = assemble(transpiled_circuit, shots=1000, seed_simulator=self.seed)
+        logger = getLogger()
+        logger.setLevel('DEBUG')
+        self.log_output = io.StringIO()
+        logger.addHandler(StreamHandlerRaiseException(self.log_output))
+
+    def assertExecuteLog(self, log_msg):
+        """ Runs execute and check for logs containing specified message"""
+        shots = 100
+        qr = QuantumRegister(2, 'qr')
+        cr = ClassicalRegister(4, 'cr')
+        circuit = QuantumCircuit(qr, cr)
+        execute(
+            circuit,
+            backend=self.backend,
+            shots=shots,
+            seed_simulator=self.seed)
+        self.log_output.seek(0)
+        # Filter unrelated log lines
+        output_lines = self.log_output.readlines()
+        execute_log_lines = [x for x in output_lines if x.__contains__(log_msg)]
```

Comment:
Is there a reason not to do:

```suggestion
        transpile_log_lines = [x for x in output_lines if log_msg in x]
```
Calling the private methods seems a bit weird.